tests/libtest.sh

```diff
@@ -18,6 +18,10 @@
 # Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 # Boston, MA 02111-1307, USA.
 
+# redirect stdout to stderr, otherwise the log will have command output out of
+# order with xtrace output
+exec 2>&1
```

Comment:
Is this really correct? The test logs are in TAP format, for which stdout carries a machine-readable log.

Reply:
Sorry, had it backwards, should say "redirect stderr to stdout". It is hard to see how that could cause a problem with the test harness, since stderr shouldn't have strings in the TAP format ("ok ...", "not ok ...", etc.), and I couldn't find another way to get the output to stay in order.

Reply:
This is going to become an issue if we want to move to Meson, because Automake has a permissive TAP parser (any unrecognised string is treated like a `# diagnostic`), but Meson has a strict TAP parser (any unrecognised string is an error).

Reply:
Are you sure? The spec [here](https://testanything.org/tap-specification.html) says "A TAP parser is required to not consider an unknown line as an error but may optionally choose to capture said line and hand it to the test harness, which may have custom behavior attached."

Reply:
And don't we already have non-TAP lines in the output, everything else produced on stdout?

Reply:
Yes, I'm sure Meson parses strictly. I tried porting bubblewrap to Meson, and it failed tests until I applied https://github.com/containers/bubblewrap/commit/0d88fe5e60f08b8afbd2f2546e5f436871aa5112. Look for `unexpected input at line` in <https://github.com/mesonbuild/meson/blob/master/mesonbuild/mtest.py>.

It's entirely possible that Meson is so strict as to be non-TAP-spec-compliant, although I'm sure its authors would argue that making the test fail is *technically* an acceptable example of "custom behavior".

> don't we already have non-TAP lines in the output, everything else produced on stdout?

In theory that's nothing, because in theory we don't/shouldn't write anything unstructured to stdout - although I don't know how true that is in reality, and I suspect we won't find out unless/until we try parsing tests' output with a strict TAP parser.

Reply:
I agree it's bad that stdout and stderr go so dramatically out-of-sync. I've experimented with `stdbuf(1)` in the past (e.g. https://salsa.debian.org/debian/ostree/-/commit/eeb52cb9f60f7ed5032c7d00d54137d9622b6cb8 in the Debian packaging of `libostree`) which might be a reasonable longer-term approach?

Reply:
I actually already tried prepending the flatpak commands run by the tests with `stdbuf -o0` and for some reason that didn't solve the out-of-sync problem.

Reply:
I wonder if another solution would be to redirect the output of the flatpak commands run by the tests to stderr, since I think that's the main thing we currently write to stdout. On the other hand we are not going to be moving to meson any time soon (https://github.com/flatpak/flatpak/issues/2241)